### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by for-ashraf**
In this PR, We have incorporated the following requirements:
Config & basic setup
Create React App.
Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
Create routes and view components (Rockets, Missions, My Profile, Dragons [only if your team has 3 members]). Use <NavLink /> for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
Create directories for all Redux state slice files (rockets, missions, dragons [only if your team has 3 members]).
Redux: Fetch data and update Redux store
Upon first render fetch data from the SpaceX API endpoints:

Rockets: https://api.spacexdata.com/v3/rockets
Missions: https://api.spacexdata.com/v3/missions
Dragons: https://api.spacexdata.com/v3/dragons [only if your team has 3 members]
Once the data are fetched, dispatch an action to store the selected data in Redux store:

Rockets:
id
rocket_name
description
flickr_images
Missions:
mission_id
mission_name
description
Dragons [only if your team has 3 members]:
id
name
type
flickr_images
NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

NOTE: Rockets is the default view, so you must fetch rockets data when the application starts. However, the missions data should only be fetched (once) when a user navigates to the Missions section.

Render UI:lists
Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. i.e.:
// get rockets data from the store
const rockets = useSelector(state => state.rockets);
You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
Render a list of rockets (as per design). For the image of a rocket use the first image in the array of flickr_images.
Render a table with the missions' data (as per design).
Render a list of dragons (as per design). For the image of a dragon use the first image in the array of flickr_images[only if your team has 3 members].
Redux: Write actions and reducers for booking rockets/dragons and joining missions
When a user clicks the "Reserve rocket" button or "Reserve dragon" button [only if your team has 3 members], action needs to be dispatched to update the store. You need to get the ID of the reserved rocket and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all rockets, but the selected rocket will have an extra key reserved with its value set to true. You could use a JS filter() or map() to set the value of the new state - i.e.:
const newState = state.map(rocket => {
    if(rocket.id !== id) 
        return rocket;
    return { ...rocket, reserved: true };
});
Regardless of which method you choose, make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct rocket ID as an argument.
Create a reducer and action dispatcher for the "Join Mission" button. The logic here is practically the same as with rockets - you need to pass the mission's ID to the corresponding action and update the missions' state with the selected mission having a new key/value - reserved: true.
Redux: Write actions and reducers for canceling rockets/dragons and leaving missions
Here you need to follow the same logic as with the "Reserve rocket"/"Reserve dragon" and "Join mission" - but you need to set the reserved key to false.
Dispatch these actions upon click on the corresponding buttons.
Render UI: conditional components rendering
Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design) .
Dragons that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve dragon" (as per design) [only if your team has 3 members].
Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).
Rockets/Dragons and Missions should use the React conditional rendering syntax:
{rocket.reserved && ( 
    // render Cancel Rocket button
)}
Render UI: My Profile section
Compose two/three column layout and list ONLY the rockets/dragons reserved and missions joined by the user (as per design):
Render a list of all joined missions (use filter()).
Render a list of all reserved rockets (use filter()).
Render a list of all reserved dragons (use filter()) [only if your team has 3 members]_.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import Header from './components/header';
+import Profile from './pages/Profile';
+import Rockets from './pages/Rockets';
```
Comment (MeqdamAlqudah) [inline src/App.js:4]: - [ ] Please add some tests to the application.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import Missions from './pages/Missions';
+import './App.css';
+
```
Comment (MeqdamAlqudah) [inline src/App.js:7]: - [ ] Please add Redux logger to redux and make sure it is working 
image

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+function App() {
+  return (
```
Comment (MeqdamAlqudah) [inline src/App.js:9]: - [Optional] Please make sure to save all the data (Reserved missions, Rockest....etc) in the localStorage

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @for-ashraf @diegoyon 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:+1: Readme file

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @for-ashraf , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Let me point out:
- App works as expected 👍🏻 
- Tests are passing successfully 👍🏻 
- Professional documentation

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**